asserts/model.go

```diff
@@ -657,6 +694,117 @@ func checkOptionalSystemUserAuthority(headers map[string]interface{}, brandID st
+	accountID, err := checkOptionalStringWhat(headers, "account-id", what)
+	if err != nil {
+		return "", err
+	}
+
+	// default to brand ID if account ID is not provided
+	if accountID == "" {
+		return brandID, nil
+	}
+	return accountID, nil
+}
+
+// checkModelValidationSetSequence reads the optional 'sequence' member, if
+// not set, returns 0 as this means unpinned. Unfortunately we are not able
+// to reuse `checkSequence` as it operates inside different parameters.
+func checkModelValidationSetSequence(headers map[string]interface{}, what string) (int, error) {
+	// Default to 0 when the sequence header is not present
+	if _, ok := headers["sequence"]; !ok {
+		return 0, nil
+	}
+
+	seq, err := checkIntWhat(headers, "sequence", what)
+	if err != nil {
+		return 0, err
+	}
+
+	// If sequence is provided, only accept positive values above 0
+	if seq <= 0 {
+		return 0, fmt.Errorf("\"sequence\" %s must be larger than 0 or left unspecified (meaning tracking latest)", what)
+	}
+	return seq, nil
+}
+
+func checkModelValidationSetMode(headers map[string]interface{}, what string) (ModelValidationSetMode, error) {
+	modeStr, err := checkNotEmptyStringWhat(headers, "mode", what)
+	if err != nil {
+		return "", err
+	}
+
+	if modeStr != "" && !strutil.ListContains(validModelValidationSetModes, modeStr) {
+		return "", fmt.Errorf("\"mode\" %s must be %s, not %q", what, strings.Join(validModelValidationSetModes, "|"), modeStr)
+	}
+	return ModelValidationSetMode(modeStr), nil
+}
+
+func checkModelValidationSet(headers map[string]interface{}, brandID string) (*ModelValidationSet, error) {
+	name, err := checkStringMatchesWhat(headers, "name", "of validation-set", validValidationSetName)
+	if err != nil {
+		return nil, err
+	}
+
+	what := fmt.Sprintf("of validation-set %q", name)
+	accountID, err := checkModelValidationSetAccountID(headers, what, brandID)
+	if err != nil {
+		return nil, err
+	}
+
+	what = fmt.Sprintf("of validation-set \"%s/%s\"", accountID, name)
+	seq, err := checkModelValidationSetSequence(headers, what)
+	if err != nil {
+		return nil, err
+	}
+
+	mode, err := checkModelValidationSetMode(headers, what)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ModelValidationSet{
+		AccountID: accountID,
+		Name:      name,
+		Sequence:  seq,
+		Mode:      mode,
+	}, nil
+}
+
+func checkOptionalModelValidationSets(headers map[string]interface{}, brandID string) ([]*ModelValidationSet, error) {
+	valSets, ok := headers["validation-sets"]
+	if !ok {
+		return nil, nil
+	}
+
+	entries, ok := valSets.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf(`"validation-sets" must be a list of validation sets`)
+	}
+
+	vss := make([]*ModelValidationSet, len(entries))
+	seen := make(map[string]bool, len(entries))
+	for i, entry := range entries {
+		data, ok := entry.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf(`"validation-sets" must contain a list of validation sets`)
```

Comment:
The difference from above error is subtle and perhaps potentially little confusing. Perhaps consider something like "Every validation set must contain a list of properties"?